### PR TITLE
fix(ui): handle "None" workaround flag for IPMI power field

### LINKED ISSUE #3523: IPMI Workaround "None" option should not be displayed and instead set when all other options are unset
**Describe the bug**
In the multiselect for IPMI workaround options, there is a `None` option displayed, the explicit `None` value is used so that the CLI may unset the default value. However, the UI should hide this option and set it as a value when all other options are unset.

**maas-ui version**
HEAD

**MAAS version**
HEAD

**LP issue**
lp1958173

**To Reproduce**
1. Add a machine with IPMI power management
2. Configure IPMI options
3. Notice the `None` workaround option

**Expected behavior**
All options except `None` are displayed and available for selection, if all of the displayed options are unset, the `None` option is selected.

**Screenshots**

Screenshot from 2022-01-17 14-44-37

**Desktop (please complete the following information):**
 - OS: Ubuntu 21.10
 - Browser Firefox
 - Version HEAD

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Created a custom component for IPMI power fields which handles the "None" workaround flag

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list and click "Add hardware > Machine"
- Select "IPMI" as the power type and check there is no "None" workaround flag
- Unselect all workaround flags and submit the form
- Check that the action that gets dispatched includes the power parameter `workaround_flags: [""]` (where an empty string is the value for "None")
- Add another IPMI machine, this time with workaround flags
- Check that the action that gets dispatched does not include the "None" value

## Fixes

Fixes #3523

### CODE REVIEW

File: ui/src/app/base/components/PowerTypeFields/IPMIPowerFields/IPMIPowerFields.test.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      onSubmit={jest.fn()}
+    >
+      <IPMIPowerFields fields={[workaroundsField]} />
+    </Formik>
+  );
+
+  await waitFor(() => {
+    expect(
+      screen.queryByRole("checkbox", { name: "None" })
+    ).not.toBeInTheDocument();
```
Comment (huwshimi) [inline ui/src/app/base/components/PowerTypeFields/IPMIPowerFields/IPMIPowerFields.test.tsx:41]: It might be useful to also check the "One" checkbox exists to help catch that this isn't just a bad queryBy.

File: ui/src/app/base/components/PowerTypeFields/IPMIPowerFields/IPMIPowerFields.test.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    label: "Workaround flags",
+    name: WORKAROUNDS_FIELD_NAME,
+    choices: [
+      ["one", "One"],
+      [NONE_WORKAROUND_VALUE, "None"],
+    ],
+  });
+});
+
+it("does not render the 'None' choice for the workaround flags field", async () => {
```
Comment (huwshimi) [inline ui/src/app/base/components/PowerTypeFields/IPMIPowerFields/IPMIPowerFields.test.tsx:26]: I think it's probably also worth adding tests for when it's not a multi-choice and also for selecting the "None" choice when all other fields are unselected.

**Review by huwshimi — APPROVED** [review]
A couple of comments around the tests.
